algorithms/alg.verify-event-nipopow.tex

```diff
@@ -0,0 +1,17 @@
+\begin{algorithm}[H]
+    \caption{\label{alg.verify-event-nipopow} A \textsf{verify} implementation which verifies a spend event on the bitcoin blockchain using NIPoPoWs.}
+    \begin{algorithmic}[1]
+        \Function{$Q_b$}{$S$}
+            \State\Return{$b \in S$}
+        \EndFunction
+        \Function{$\mathsf{verify}_{k,m}^{\mathcal{G},e}$}{$\alpha, \mathcal{P}$}
+            \If{$\neg \textsf{verify-tx}_e(\alpha)$}
+                \State\Return{$\false$}
+            \EndIf
+
+            \Let{blk}{\textsf{decodeBlockHeader}(\alpha.\textsf{blockHeader})}
+            \State\Return{$\textsf{verify-infix}_{m,k}^{\mathcal{G},Q_\text{blk}}(\mathcal{P})$}
```

Comment:
We need to say a few words in the text about what $verify-infix$ is. At this point I would also avoid using $m$ and $k$ as they do not contribute to the discussion and we are not introducing them. Furthermore, you are already using $m$ for the federation size.